pipeline/03-evaluate.R

```diff
@@ -75,7 +75,11 @@ gen_agg_stats <- function(data, truth, estimate, bldg_sqft,
                           class, col_dict, min_n) {
   # Helper function to return NA when sale sample size is too small
   gte_n <- \(n_sales, min_n, fn, na_type) {
-    ifelse(sum(!is.na(n_sales)) >= min_n, fn, na_type)
+    if (sum(!is.na(n_sales)) >= min_n) {
+      return(fn)
+    } else {
+      return(rep(na_type, length(n_sales)))
+    }
```

Comment:
The `fn` bit in the `ifelse()` version of this code is still evaluated, even if the condition is `FALSE`. That means if the `fn` throws an error for small samples (i.e. `prb()`), then the whole pipeline fails. This version prevents code execution of the second branch if the sample size is too small.

Reply:
Very annoying behavior on R's part 😓